ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py

```diff
@@ -2766,42 +2769,51 @@ def test_numpy_ndarray_max(
     class_tree=CLASS_TREE,
     init_tree="numpy.array",
     method_name="mean",
-    dtype_x_axis=helpers.dtype_values_axis(
-        available_dtypes=helpers.get_dtypes("float"),
-        min_axis=-1,
-        max_axis=0,
-        min_num_dims=1,
-        force_int_axis=True,
-    ),
+    dtype_and_x=_statistical_dtype_values(function="mean"),
+    dtype=helpers.get_dtypes("float", full=False, none=True),
+    where=np_frontend_helpers.where(),
+    keep_dims=st.booleans(),
 )
 def test_numpy_ndarray_mean(
-    dtype_x_axis,
+    dtype_and_x,
+    dtype,
+    where,
+    keep_dims,
     frontend_method_data,
     init_flags,
     method_flags,
     backend_fw,
     frontend,
     on_device,
 ):
-    input_dtypes, x, axis = dtype_x_axis
+    input_dtypes, x, axis = dtype_and_x
+    where, input_dtypes, test_flags = np_frontend_helpers.handle_where_and_array_bools(
+        where=where,
+        input_dtype=input_dtypes,
+        test_flags=method_flags,
+    )
+    if "bfloat16" in input_dtypes + dtype or []:
```

Comment:
Ideally we should filter `bfloat16` in `@handle_frontend_method`.

Reply:
Earlier due to some issues, `bfloat16` was getting generated for `numpy` as well, but it looks like it has been resolved now. so will remove it